includes/data-stores/class-wc-order-item-data-store.php

```diff
@@ -63,6 +69,8 @@ public function update_order_item( $item_id, $item ) {
 	 * @param  int $item_id Item ID.
 	 */
 	public function delete_order_item( $item_id ) {
+		$this->clear_cache( $item_id, null );
```

Comment:
I put the cache clear before the delete statements because the `$order_id` lookup done in the `clear_cache()` method would fail otherwise, leaving that cache uncleared. In hindsight though, there's an easy way to avoid this pitfall.